### PR TITLE
Result passing property added for other projects.

### PR DESCRIPTION
**Pull request by crcrewso**
The intent of this patch is to create a easy to maintain self building dictionary, that can be called by external classes or frameworks. 

Basic format would be:
```
analyzed_cbct[<module>][<result type or sub module>]
```

I.e. 
```
analyzed_cbct['CTP404']['HUs']['Air'] 
```
would store the analyzed measure for the air volume of the CTP404 module. 

At all points the indices would be human readable, and for modules with more or less roi's then the function could be redefined from the inherited module. 

Documentation of the human readable names would have to be added to the readthedocs of course.

### CODE REVIEW

File: pylinac/ct.py [hunk 1]
```diff
@@ -1246,6 +1287,19 @@ def results(self):
...
+    def results_dict(self):
+        result = {}
+        if self._has_module(CTP404CP503):
+            result["CTP404"] = self.ctp404.results_dict
+        if self._has_module(CTP486):
+            result["CTP486"] = self.ctp486.results_dict
+        if self._has_module(CTP515):
+            result["CTP515"] = self.ctp515.results_dict
+        if self._has_module(CTP528CP504):
+            result["CTP528"] = self.ctp528.result_dict
```
Comment (jrkerns) [inline pylinac/ct.py:1301]: typo here

**Comment by jrkerns** [comment]
Thanks for the pull request. If I'm reading this right, you'll need to actually call `myct.results_dict['CTP404']...`, correct? 

If indeed this is what you want, you'll need to remove this property from the 604 as it's inherited from `CatPhanBase` and generalize the `CatPhanBase` method to handle all modules, not just ones from specific phantoms (i.e. it can't have `<module>CP604`) and add unit tests (probably [here](https://github.com/jrkerns/pylinac/blob/master/tests_basic/test_cbct.py#L43)) or per-phantom (probably [here](https://github.com/jrkerns/pylinac/blob/master/tests_basic/test_cbct.py#L200)) and update the docs (I'd put it in the "view the results" section [here](https://pylinac.readthedocs.io/en/latest/cbct_docs.html#typical-use).

**Comment by crcrewso** [comment]
> _Corrected_:
> Thanks for the pull request. If I'm reading this right, you'll need to actually call `myct.results_dict['CTP404']...`, correct?
> 
> If indeed this is what you want, you'll need to add unit tests (probably [here](https://github.com/jrkerns/pylinac/blob/master/tests_basic/test_cbct.py#L43)) or per-phantom (probably [here](https://github.com/jrkerns/pylinac/blob/master/tests_basic/test_cbct.py#L200)) and update the docs (I'd put it in the "view the results" section [here](https://pylinac.readthedocs.io/en/latest/cbct_docs.html#typical-use).

Yes, I do intend to support all phantoms in the same way, creating a unified consistent jsonable object for any cbct analysis. Will amend this pull request as I have time.

**Comment by jrkerns** [comment]
Good news on this front. I'm working on adding integration via APIs, so a dict of results (converted to JSON) can be helpful here. I'll tackle this once I can unify the workflow.

**Comment by jrkerns** [comment]
Closed in #360

**Pull request closed without merging**